### PR TITLE
Forms

### PR DESCRIPTION
**Pull request by tingamapuro04**
#### n+1
- Make sure that the N+1 problem is solved when fetching all posts and their comments for a user.

#### Integration specs
-  Use Capybara to write integration tests for each view in your project.
  
- User index page:
  - I can see the username of all other users.
  - I can see the profile picture for each user.
  - I can see the number of posts each user has written.
  - When I click on a user, I am redirected to that user's show page.
  
- user show page:
  - I can see the user's profile picture.
  - I can see the user's username.
  - I can see the number of posts the user has written.
  - I can see the user's bio.
  - I can see the user's first 3 posts.
  - I can see a button that lets me view all of a user's posts.
  - When I click a user's post, it redirects me to that post's show page.
  - When I click to see all posts, it redirects me to the user's post's index page.

- User post index page:
  - I can see the user's profile picture.
  - I can see the user's username.
  - I can see the number of posts the user has written.
  - I can see a post's title.
  - I can see some of the post's body.
  - I can see the first comments on a post.
  - I can see how many comments a post has.
  - I can see how many likes a post has.
  - I can see a section for pagination if there are more posts than fit on the view.
  - When I click on a post, it redirects me to that post's show page.

- Post show page:
  - I can see the post's title.
  - I can see who wrote the post.
  - I can see how many comments it has.
  - I can see how many likes it has.
  - I can see the post body.
  - I can see the username of each commentor.
  - I can see the comment each commentor left.

**Be sure to include integration specs for any other features you may have implemented!**
    [Read More](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/projects/intergration_spec_and_n_1_project.md)

### CODE REVIEW

File: app/controllers/posts_controller.rb [hunk 1]
```diff
@@ -1,4 +1,13 @@
...
+  def skip_bullet
+    Bullet.enable = false
+    yield
+  ensure
+    Bullet.enable = true
+  end
+
   def index
     @user2 = User.find(params[:user_id])
     @posts = @user2.posts
```
Comment (FrederickMih) [inline app/controllers/posts_controller.rb]: - [ ] Let's keep our application scalable by eliminating the N+1 query problems from our application(Seems n extra queries are being executed when fetching all posts and comments for a @user2) Kindly fix these issues.

**Hints:**  We can try to eliminate the issue like this:

@posts = @user2.posts.includes(:comments)

**Review by FrederickMih — CHANGES_REQUESTED** [review]
Hi @tingamapuro04, 

Good job so far!
You still need to work on some issues to go to the next project, but you are almost there!



## Highlights :green_circle: 

- Correct Gitflow :heavy_check_mark: 
- PR has a title and summary :heavy_check_mark: 


## Required Changes ♻️

- [ ] Kindly remove the instructions in the Getting Started section of your README.md file. They are directives so you were not supposed to include them again in your final READMD.md file.

_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by tingamapuro04** [comment]
I have included the suggested correction

**Review by omarsalem7 — APPROVED** [review]
Hi @tingamapuro04 😊, 
# STATUS: APPROVED! 🥇 💯 

giphy

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 🚀 
Congratulations! 🎉 

## Highlights
✅ Project has met all the technical aspects you needed for this project. Keep rocking! 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**